CenterForOpenScience/osf.io
# [ENG-4642] Boa Add-on Project PR - OSF Part

## Purpose

Implement a new add-on for Boa to submit Boa query file from OSF projects and download results.

### Note:

* Migration required
* Ember part: https://github.com/CenterForOpenScience/ember-osf-web/pull/2046
* Release note: https://www.notion.so/cos/Boa-Add-on-Release-372fe3b1b5dd42849f6c1c06536c3768

## Changes

### Primary

* New add-ons `boa`: model, serializer, views, exceptions
* Submit to boa
  * A button in project widget and a pop-up modal for user to select dataset and submit to boa
  * A view to handle submission from legacy and ember FE
  * A task for the view to call. It downloads the Boa query file, submit the query to Boa, polling for results and download the results. It also sends email notifications on success/failure
* Expanded `_collect_addons()` to support active add-ons
* Added requirements to use Boa API and to support async celery and async django
* Unit tests

### Other

* Added test requirements to support mocking and testing async
* Improved base tests for add-ons
* Fixed a bug in `owncloud` add-on
* Rewrote `owncloud` add-on tests

## QA Notes

See [QA Test Plan for Boa](https://docs.google.com/document/d/1XuiSKV2JPpPdXt8aqFXOfBvMXSIA2qz8a1Bxevqh6-M/edit)

## Documentation

N/A

## Side Effects

N/A

## Ticket

https://openscience.atlassian.net/browse/ENG-4642
